pyopendds/dev/include/pyopendds/user.hpp

```diff
@@ -234,6 +239,34 @@ class TopicType : public TopicTypeBase {
     return rv;
   }
 
+  PyObject* write(PyObject* pywriter, PyObject* pysample)
+  {
+    DDS::DataWriter* writer = get_capsule<DDS::DataWriter>(pywriter);
+    if (!writer) PyErr_SetString(PyExc_Exception, "writer is a NULL pointer");
```

Comment:
```suggestion
    if (!writer) throw Exception();
```

So `PyErr_SetString` doesn't stop this function from running. All it does is set up an exception that the interrupter will throw when it has the chance. Also [`get_capsule` is already setting `PyErr_SetString`](https://github.com/oci-labs/pyopendds/blob/f667bae05e038a1194740c0b92c7356bbd49a599/pyopendds/dev/include/pyopendds/common.hpp#L87-L88), all we would have to do here is to stop running this function, which is the purpose of throwing `Exception()` (with no arguments).

Reply:
done